File: twoSum.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> m;
        for(int i=0;i<nums.size();i++) {
            int t=target-nums[i];
            if(m.find(t) == m.end()) {
                m[nums[i]]=i;
            } else {
                result.push_back(m[t]);
                result.push_back(i);
                return result;
            }
        }
        return result;
    }
};

File: java/TwoSum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> m = new HashMap<Integer,Integer>();
        for(int i=0;i<nums.length;i++) {
            m.put(target-nums[i],i);
        }
        for(int i=0;i<nums.length;i++) {
            Integer v = m.get(nums[i]);
            if(v!=null && v!=i) {
                return new int[]{i,v};
            }
        }
        throw new RuntimeException();
    }
}
